src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -260,6 +277,24 @@ public void periodic() {
                         chassisSpeeds.vyMetersPerSecond = yPID.calculate(getPose().getTranslation().minus(translationTarget).getY());
                 }
 
+                if(balanceTarget != null ) {
+                        if (Math.abs(getGyroscopePitch().getDegrees()) >= AUTO_BALANCE_TOLERANCE_DEGREES) {
+                                chassisSpeeds.vyMetersPerSecond = yPID.calculate(getGyroscopePitch().minus(balanceTarget).getRadians());
+                                totalMetersTranslated += chassisSpeeds.vyMetersPerSecond * (Timer.getFPGATimestamp() - lastUpdateTimeBalanceTarget);
+                                lastUpdateTime = Timer.getFPGATimestamp();
+                        } else if (Math.abs(getGyroscopeRoll().getDegrees()) >= AUTO_BALANCE_TOLERANCE_DEGREES) {
+                                chassisSpeeds.vxMetersPerSecond = xPID.calculate(getGyroscopeRoll().minus(balanceTarget).getRadians());
+                                totalMetersTranslated += chassisSpeeds.vxMetersPerSecond * (Timer.getFPGATimestamp() - lastUpdateTimeBalanceTarget);
+                                lastUpdateTime = Timer.getFPGATimestamp();
```

Comment:
Should this be last updateTimeBalanceTarget? You could also pull this line of code out of the if else as its the same for both paths.